### ISSUE TITLE
JavaVersion does not handle Java 13

### ISSUE THREAD

**Issue by snicoll** [hazelcast/hazelcast#15348]
In the process of adding a compatibility build for Java 13, one of our hazelcast integration tests started failing:

```
Caused by: com.hazelcast.config.InvalidConfigurationException: Invalid YAML configuration
 	at com.hazelcast.client.config.YamlClientConfigBuilder.parseAndBuildConfig(YamlClientConfigBuilder.java:138)
 	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:125)
 	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:114)
 	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:109)
 	at com.hazelcast.client.impl.clientside.FailoverClientConfigSupport.locateAndCreateClientConfig(FailoverClientConfigSupport.java:130)
 	at com.hazelcast.client.impl.clientside.FailoverClientConfigSupport.resolveClientConfig(FailoverClientConfigSupport.java:91)
 	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:115)
 	at org.springframework.boot.autoconfigure.hazelcast.HazelcastClientConfiguration$HazelcastClientConfigFileConfiguration.hazelcastInstance(HazelcastClientConfiguration.java:57)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
 	... 90 more
 Caused by: com.hazelcast.internal.yaml.YamlException: An error occurred while loading and parsing the YAML stream
 	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:74)
 	at com.hazelcast.client.config.YamlClientConfigBuilder.parseAndBuildConfig(YamlClientConfigBuilder.java:136)
 	... 102 more
 Caused by: java.lang.UnsupportedOperationException: Processing YAML documents requires Java 8 or higher version
 	at com.hazelcast.internal.yaml.YamlUtil.ensureRunningOnJava8OrHigher(YamlUtil.java:116)
 	at com.hazelcast.internal.yaml.YamlLoader.getLoad(YamlLoader.java:157)
 	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:69)
 	... 103 more
```

[Parsing the system property](https://github.com/hazelcast/hazelcast/blob/80688172ec0cf60f40ca104cf2fbaacc8f9b500c/hazelcast/src/main/java/com/hazelcast/internal/util/JavaVersion.java#L57) is fragile, please consider alternatives and a less agressive outcome when the Java version can't be determined.

**Comment by philwebb** [comment]
[This code](https://github.com/spring-projects/spring-boot/blob/38a42a86008460d20b8df2c1f7a997b6e47824dd/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/system/JavaVersion.java) might help if you're interested in taking a different approach.

**Comment by mmedenjak** [comment]
Merged for maintenance (3.12.x), we have an issue for master to improve the situation in 4.0. Thank you for the report @snicoll !
